src/tests/test_sexual_behaviour.py

```diff
@@ -1,22 +1,58 @@
+from datetime import date
+
 import numpy as np
 
-from hivpy.sexual_behaviour import (prob_transition,
-                                    sex_behaviour_trans_female,
-                                    sex_behaviour_trans_male)
+from hivpy import sex_behaviour_data as sbd
+from hivpy.demographics import SexType
+from hivpy.population import Population
+from hivpy.sexual_behaviour import SexualBehaviourModule
 
 
-def check_prob_sums(gender, trans_matrices):
-    (n_mat, dim, _) = trans_matrices.shape
+def check_prob_sums(sex, trans_matrix):
+    SBM = SexualBehaviourModule()
+    SBM.sex_behaviour_trans[sex] = trans_matrix
+    (dim, _) = trans_matrix.shape
     for i in range(0, dim):
-        for age in [15, 20, 25, 30, 35, 40, 45, 50, 55, 60, 65]:
-            tot_prob = 0
-            for j in range(0, dim):
-                tot_prob += prob_transition(gender, age, i, j)
-            assert(np.isclose(tot_prob, 1.0))
+        ages = np.array([15, 20, 25, 30, 35, 40, 45, 50, 55, 60, 65])
+        tot_prob = np.array([0.0]*len(ages))  # probability for each age range
+        for j in range(0, dim):
+            tot_prob += SBM.prob_transition(sex, ages, i, j)
+        assert(np.allclose(tot_prob, 1.0))
 
 
 def test_transition_probabilities():
-    assert (sex_behaviour_trans_male.shape == (15, 4, 4))
-    assert (sex_behaviour_trans_female.shape == (15, 2, 2))
-    check_prob_sums(1, sex_behaviour_trans_female)
-    check_prob_sums(0, sex_behaviour_trans_male)
+    for trans_matrix in sbd.sex_behaviour_trans_male_options:
+        assert (trans_matrix.shape == (4, 4))
+        check_prob_sums(SexType.Male, trans_matrix)
+    # for trans_matrix in sbd.sex_behaviour_trans_female_options:
+    #     assert (trans_matrix.shape == (2, 2))
+    #     check_prob_sums(1, trans_matrix)
```

Comment:
The commented-out tests also pass for me (changing 1 to `SexType.Female`)!

Reply:
I will put it back in properly fixed; it might just be that I broke the test while rewriting the functions!